ipatests/test_integration/test_dns_locations.py

```diff
@@ -186,6 +186,20 @@ def _test_A_rec_against_server(self, server_ip, domain, expected_servers,
                 "with IP: '{}' for name '{}' (expected:\n{}\ngot:\n{})".
                 format(server_ip, name_abs, expected, query))
 
+    def _test_AAAA_rec_against_server(self, server_ip, domain,
+                                      expected_servers,
+                                      rec_list=IPA_CA_A_REC):
+        for rname in rec_list:
+            name_abs = rname.derelativize(domain)
+            expected = _gen_expected_a_rrset(name_abs, expected_servers)
```

Comment:
_gen_expected_a_rrset() is hardcoding "dns.rdatatype.A" as RdataType when returning dns.rrset.from_text_list(), I wonder if this should be a new param, then you could use it for both use cases A and AAAA rdatatypes.